ext/node/polyfills/internal/buffer.mjs

```diff
@@ -204,20 +221,42 @@ Buffer.allocUnsafeSlow = function allocUnsafeSlow(size) {
   return _allocUnsafe(size);
 };
 
+function fromStringFast(string, ops) {
+  const length = ops.byteLength(string);
+  if (length >= (Buffer.poolSize >>> 1)) {
+    const data = ops.create(string);
+    Object.setPrototypeOf(data, Buffer.prototype);
+    return data;
+  }
+
+  if (length > (poolSize - poolOffset)) {
+    createPool();
+  }
+  let b = new Uint8Array(allocPool, poolOffset, length);
+  Object.setPrototypeOf(b, Buffer.prototype);
```

Comment:
Possibly `let b = Reflect.construct(Uint8Array, [allocPool, poolOffset, length], Buffer);` is better for performance.

Reply:
Node.js sets up the prototype in this way which makes it so that `new FastBuffer` creates a `Buffer` instance directly without overhead involved with changing the prototype.

```js
// lib/internal/buffer.js
class FastBuffer extends Uint8Array {
  // Using an explicit constructor here is necessary to avoid relying on
  // `Array.prototype[Symbol.iterator]`, which can be mutated by users.
  // eslint-disable-next-line no-useless-constructor
  constructor(bufferOrLength, byteOffset, length) {
    super(bufferOrLength, byteOffset, length);
  }
}

// lib/buffer.js
FastBuffer.prototype.constructor = Buffer;
Buffer.prototype = FastBuffer.prototype;
```

(The explicit constructor is no longer necessary since V8 12.1.)